tests/test_structured_state.py

```diff
@@ -744,3 +744,94 @@ def test_measure_circ(circuit: Circuit) -> None:
     # Check sample frequency consistent with theoretical probability
     for outcome, count in sample_dict.items():
         assert np.isclose(count / n_samples, p[outcome], atol=0.1)
+
+
+def test_mps_qubit_addition_and_measure() -> None:
+    with CuTensorNetHandle() as libhandle:
+        config = Config()
+        mps = MPSxGate(
+            libhandle,
+            qubits=[Qubit(0), Qubit(1), Qubit(2), Qubit(3)],
+            config=config,
+        )
+
+        x = cp.asarray(
+            [
+                [0, 1],
+                [1, 0],
+            ],
+            dtype=config._complex_t,
+        )
+        cx = cp.asarray(
+            [
+                [1, 0, 0, 0],
+                [0, 1, 0, 0],
+                [0, 0, 0, 1],
+                [0, 0, 1, 0],
+            ],
+            dtype=config._complex_t,
+        )
+
+        # Apply some gates
+        mps.apply_unitary(x, [Qubit(1)])  # |0100>
+        mps.apply_unitary(cx, [Qubit(1), Qubit(2)])  # |0110>
+        mps.apply_unitary(cx, [Qubit(2), Qubit(3)])  # |0111>
+        # Add a qubit at the end of the MPS
+        mps.add_qubit(new_qubit=Qubit(4), position=len(mps))  # |01110>
+        # Apply some more gates acting on the new qubit
+        mps.apply_unitary(cx, [Qubit(3), Qubit(4)])  # |01111>
+        mps.apply_unitary(cx, [Qubit(4), Qubit(3)])  # |01101>
+        # Add a qubit at position 3
+        mps.add_qubit(new_qubit=Qubit(6), position=3)  # |011001>
+        # Apply some more gates acting on the new qubit
+        mps.apply_unitary(x, [Qubit(6)])  # |011101>
+        mps.apply_unitary(cx, [Qubit(6), Qubit(2)])  # |010101>
+        mps.apply_unitary(cx, [Qubit(6), Qubit(3)])  # |010111>
+        # Add another qubit at the end of the MPS
+        mps.add_qubit(new_qubit=Qubit(5), position=len(mps), state=1)  # |0101111>

```

Comment:
Have you added a test of adding qubits somewhere in the middle?

Reply:
Yeah, line 785 adds a qubit in the 3rd position of the MPS.